File: 15649/Main.java
import java.io.*;
import java.util.*;

public class Main {
    static int[] arr;
    static boolean[] visit;
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String [] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N, M;
        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());

        arr = new int[M];
        visit = new boolean[N];

        dfs(N, M, 0);

        bw.flush();
        bw.close();
    }

    public static void dfs(int N, int M, int depth) throws IOException {
        if (depth == M) {
            for (int val : arr) {
                bw.write(val + " ");
            }
            bw.write("\n");
            return;
        }

        for (int i = 0 ; i < N ; i++) {
            if (!visit[i]) {
                visit[i] = true;
                arr[depth] = i + 1;
                dfs(N, M, depth + 1);
                visit[i] = false;
            }
        }
    }
}

File: 15649/main.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())

ans = [0] * M
visited = [False] * N


def dfs(n, m, depth):
    if m == depth:
        for i in ans:
            print(str(i), end=' ')
        print()
        return

    for i in range(N):
        if visited[i] is False:
            visited[i] = True
            ans[depth] = i + 1
            dfs(n, m, depth + 1)
            visited[i] = False


dfs(N, M, 0)

